### PR TITLE
[FIX] point_of_sale: creation of empty draft order

### PR DESCRIPTION
**Pull request by shag-odoo**
Steps:
- Install pos_urban_piper
- Open Point of Sale.
- Configure Food Delivery Connector for one of pos
- Place a test order.
- Edit order and add a new product.
- Accept the order.
- Mark as ready the order.

Issue:
Empty draft order in the backend is created.

Fix:
- Restricting creation of empty order whose orderline is not present.

task-4147488

### CODE REVIEW

**Comment by vlst-odoo** [comment]
I agree that orders with no orderlines don't make sense in principle, but it seems that this is not the underlying issue here. Why would we we intentionally prevent that behavior ? I think that question is: why is urban piper creating orders when no one asked for new orders to be created?

**Comment by shag-odoo** [comment]
Hello @vlst-odoo 
I think this is not only urban piper's issue , same issue can be reproduce without installing pos_urban_piper.
Task ref - https://www.odoo.com/odoo/project/1737/tasks/4144884?debug=

Thanks for the suggestion and have a great day ahead :innocent:

**Comment by vlst-odoo** [comment]
> Hello @vlst-odoo I think this is not only urban piper's issue , same issue can be reproduce without installing pos_urban_piper. Task ref - https://www.odoo.com/odoo/project/1737/tasks/4144884?debug=
> 
> Thanks for the suggestion and have a great day ahead 😇

I understand, but the problem is not that we allow the creating of empty orders, but that we create empty orders in the first place.

**Comment by vlst-odoo** [comment]
Having orders with no orderlines actually makes a lot of sense. Imagine the case where we update the system of syncing of orders between the frontend and the backend such that it becomes real time. Whenever you start a new order you will want to sync that order with the backend, even though at that point the order has no lines.

**Comment by vlst-odoo** [comment]
If we have a problem of random empty orders being created, i would suggest fixing the root cause, namely the fact that we create supposedly useless orders in the first place

**Comment by caburj** [comment]
@vlst-odoo The issue can be reproduced in standard. We'll find a better way.

File: addons/point_of_sale/models/pos_order.py [hunk 1]
```diff
@@ -1043,7 +1043,7 @@ def sync_from_ui(self, orders):
 
             if existing_draft_order:
                 order_ids.append(self._process_order(order, existing_draft_order))
-            else:
+            elif order.get('lines') or order.get('payment_ids'):
```
Comment (caburj) [inline addons/point_of_sale/models/pos_order.py:1046]: The problem with this change is that if there are other changes in the order (like a state change) but the 'lines' is empty, then the order won't be processed. We can't go with this solution. We need to find another way.

**Comment by caburj** [comment]
We need to merge the fix in enterprise now so we are closing this linked community pr.

**Pull request closed without merging**